src/main/kotlin/net/ccbluex/liquidbounce/features/command/commands/client/fakeplayer/CommandFakePlayer.kt

```diff
@@ -0,0 +1,362 @@
+    private val snapshots: MutableList<PosPoseSnapshot> = ArrayList()
+
+    private val explosionBehavior: ExplosionBehavior = ExplosionBehavior()
+
+    // the entity ids of fake players shouldn't conflict with real entity ids, so they are negative
+    private var fakePlayerId = -1
+
+    fun createCommand(): Command {
+        return CommandBuilder
+            .begin("fakeplayer")
+            .hub()
+            .subcommand(spawnCommand())
+            .subcommand(removeCommand())
+            .subcommand(clearCommand())
+            .subcommand(startRecordingCommand())
+            .subcommand(endRecordingCommand())
+            .build()
+    }
+
+    private fun spawnCommand(): Command {
+        return CommandBuilder
+            .begin("spawn")
+            .parameter(
+                ParameterBuilder
+                    .begin<String>("name")
+                    .verifiedBy(ParameterBuilder.STRING_VALIDATOR)
+                    .optional()
+                    .build()
+            )
+            .handler { _, args ->
+                checkInGame()
+                spawn(args, false)
+            }
+            .build()
+    }
+
+    private fun removeCommand(): Command {
+        return CommandBuilder
+            .begin("remove")
+            .parameter(
+                ParameterBuilder
+                    .begin<String>("name")
+                    .verifiedBy(ParameterBuilder.STRING_VALIDATOR)
+                    .optional()
+                    .build()
+            )
+            .handler { command, args ->
+                checkInGame()
+
+                if (fakePlayers.isEmpty()) {
+                    throw CommandException(translation("liquidbounce.command.fakeplayer.noFakePlayers"))
+                }
+
+                val name = args.getOrNull(0)?.toString() ?: "FakePlayer"
+
+                val playersToRemove = fakePlayers.filter { fakePlayer -> fakePlayer.name.string == name }
+
+                if (playersToRemove.isEmpty()) {
+                    chat(warning((command.result("noFakePlayerNamed", name))))
+                    chat(regular(command.result("currentlySpawned")))
+                    fakePlayers.forEach { fakePlayer ->
+                        chat(regular("- " + fakePlayer.name.string))
+                    }
+
+                    return@handler
+                }
+
+                playersToRemove.forEach { fakePlayer ->
+                    world.removeEntity(fakePlayer.id, Entity.RemovalReason.KILLED)
+                    chat(
+                        regular(
+                            command.result(
+                                "fakePlayerRemoved",
+                                BigDecimal(fakePlayer.x).setScale(1, RoundingMode.HALF_UP).toDouble(),
+                                BigDecimal(fakePlayer.y).setScale(1, RoundingMode.HALF_UP).toDouble(),
+                                BigDecimal(fakePlayer.z).setScale(1, RoundingMode.HALF_UP).toDouble()
```

Comment:
You already commented that you have this rather complex code for a good reason. `round(x * 10.0) / 10.0` would also work but it might have a few edge cases thus it is good that you chose this implementation.

When people read this code they will not know why it is the way it is. If somebody comes across it might think "Isn't this just round()?" and replace it which would change the functionality. You have two options:
1. Add a comment stating what you want to do before the two occurrences
2. Or even better extract it to a function like `roundToNDecimalPlaces(x, decimalPlaces=1)` which is kind of self documenting. Additionally you can add a KDoc-Comment which states that the function `roundToNDecimalPlaces(1234.567,decimalPlaces=1)`  will return `1234.6`.